packages/@sanity/portable-text-editor/src/editor/plugins/createWithPatches.ts

```diff
@@ -78,192 +78,195 @@ export interface PatchFunctions {
-      const syncValueAfterIncomingPatches = debounce(() => syncValue(), 0, {
-        trailing: true,
-        leading: false,
-      })
-
-      // Subscribe and deal with incoming patches
-      if (incomingPatches$) {
-        debug('Subscribing to patches')
-        patchSubscription = incomingPatches$.subscribe(({patches, snapshot}) => {
-          debug('Incoming patches', patches)
-          const remotePatches = patches.filter((p) => p.origin !== 'local')
-          if (remotePatches.length !== 0) {
-            Editor.withoutNormalizing(editor, () => {
-              remotePatches.forEach((patch) => {
-                debug(`Handling remote patch ${JSON.stringify(patch)}`)
-                withoutPatching(editor, () => {
-                  withoutSaving(editor, () => {
-                    withPreserveKeys(editor, () => {
-                      try {
-                        patchToOperations(editor, patch, patches, snapshot)
-                      } catch (err) {
-                        debug('Got error trying to create operations from patch')
-                        console.error(err)
-                      }
+    if (patches$) {
+      editor.subscriptions.push(() => {
+        debug('Subscribing to patches$')
+        const sub = patches$
+          .pipe(
+            tap(({patches}) => {
+              if (patches.every((p) => p.origin === 'local')) {
+                isPending.current = false
+              }
+            }),
+            bufferUntil(() => !isPending.current),
+            concatMap((incoming) => {
+              return incoming
+            })
+          )
+          .subscribe(({patches, snapshot}) => {
+            const remotePatches = patches.filter((p) => p.origin !== 'local')
+            if (remotePatches.length !== 0) {
+              debug('Remote patches', patches)
+              Editor.withoutNormalizing(editor, () => {
+                remotePatches.forEach((patch) => {
+                  debug(`Handling remote patch ${JSON.stringify(patch)}`)
+                  withoutPatching(editor, () => {
+                    withoutSaving(editor, () => {
+                      withPreserveKeys(editor, () => {
+                        try {
+                          patchToOperations(editor, patch, patches, snapshot)
+                        } catch (err) {
+                          debug('Got error trying to create operations from patch')
+                          console.error(err)
+                        }
+                      })
                     })
                   })
                 })
               })
-            })
-          }
-          if (patches.length > 0) {
-            syncValueAfterIncomingPatches()
-          }
-        })
-      }
-      const {apply} = editor
-      editor.apply = (operation: Operation): void | Editor => {
-        if (editor.readOnly) {
-          apply(operation)
-          return editor
+            }
+            // if (patches.length > 0) {
```

Comment:
Should these be removed? If not, add a comment about why it's useful to keep?

Reply:
Should be removed. Tnx!